### PR TITLE
wofi-emoji: init at unstable-2021-05-24

### PR DESCRIPTION
**Pull request by ymarkus**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Isti115** [comment]
I tested it under NixOS and it works perfectly. :+1: 
I've been using it as a shell script for a while now, it's nice to have it packaged, thanks! :clap:

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130260` at 884a6dcf run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>wofi-emoji</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/wofi-emoji/default.nix:39:3:
```
   |
39 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/misc/wofi-emoji/default.nix:35:3:
```
   |
35 |   buildPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by ymarkus** [comment]
Added preBuild/postBuild and preInstall and postInstall hooks as suggested by @r-rmcgibbo

File: pkgs/applications/misc/wofi-emoji/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  };
+
+  nativeBuildInputs = [ jq ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    cp "${emojiJSON.src}/db/emoji.json" .
+    substituteInPlace build.sh \
+      --replace 'curl https://raw.githubusercontent.com/github/gemoji/master/db/emoji.json' 'cat emoji.json'
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/wofi-emoji/default.nix:34]: Move that to postPatch.

File: pkgs/applications/misc/wofi-emoji/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      rev = "v4.0.0.rc2";
+    };
+    meta = with lib; {
+     license = licenses.mit;
+    };
+  };
+
+in stdenv.mkDerivation rec {
+  version = "unstable-2021-05-24";
+  pname = "wofi-emoji";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/wofi-emoji/default.nix:18]:
```suggestion
  pname = "wofi-emoji";
  version = "unstable-2021-05-24";
```

File: pkgs/applications/misc/wofi-emoji/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    src = fetchFromGitHub {
+      owner = "github";
+      repo = "gemoji";
+      sha256 = "sha256-Tn0vba129LPlX+MRcCBA9qp2MU1ek1jYzVCqoNxCL/w=";
+      rev = "v4.0.0.rc2";
+    };
+    meta = with lib; {
+     license = licenses.mit;
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/wofi-emoji/default.nix:14]:
```suggestion
  emojiJSON = fetchFromGitHub {
    owner = "github";
    repo = "gemoji";
    sha256 = "sha256-Tn0vba129LPlX+MRcCBA9qp2MU1ek1jYzVCqoNxCL/w=";
    rev = "v4.0.0.rc2";
  };
```

**Comment by ymarkus** [comment]
Danke für's prüfen, @SuperSandro2000. I've changed the stuff you mentioned.

File: pkgs/applications/misc/wofi-emoji/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    owner = "github";
+    repo = "gemoji";
+    sha256 = "sha256-Tn0vba129LPlX+MRcCBA9qp2MU1ek1jYzVCqoNxCL/w=";
+    rev = "v4.0.0.rc2";
+  };
+  meta = with lib; {
+   license = licenses.mit;
+  };
+
+in stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/wofi-emoji/default.nix:14]:
```suggestion

in stdenv.mkDerivation rec {
```

**Pull request merged**